### PR TITLE
🐛(frontend) fix useUnionResource loadMore

### LINKED ISSUE #2317: "Load more" remains displayed at the bottom of the list on teacher dashboard
## Bug Report

**Problematic Behavior**

"Load more" remains displayed at the bottom of the list at `http://localhost:8070/en/dashboard/teacher/courses?query=Course`

Capture d’écran 2024-03-05 à 10 48 05

**Expected behavior/code**

It should disappear once the list is fully loaded

**Steps to Reproduce**
1. Go to `http://localhost:8070/en/dashboard/teacher/courses?query=Course`

### PR DESCRIPTION
**Pull request by rlecellier**
Because our Eof wasn't reset correctly, useUnionResource would never fetch the seconds pages and always return loadMore flag as true.

resolve #2317

### CODE REVIEW

**Review by NathanVss — APPROVED** [review]
Neat + adding a test would be nice

File: src/frontend/js/hooks/useUnionResource/utils/fetchEntity.ts [hunk 1]
```diff
@@ -65,14 +65,11 @@ export const fetchEntity = async <
...
-    // If we reached the end of the list, we set the eof flag to prevent future requests.
-    if (!res.next) {
-      eofRef.current = { ...eofRef.current, [queryKeyString]: filters.page! };
-      // Eof is cached based, the same way, we cache the fetching data. Otherwise there would
-      // be request to non existing pages after reload.
-      queryClient.setQueryData(eofQueryKey, eofRef.current);
-    }
+    // Eof is cached based, the same way, we cache the fetching data. Otherwise there would
+    // be request to non existing pages after reload.
+    const totalPages = Math.ceil(res.count / perPage);
```
Comment (NathanVss) [inline src/frontend/js/hooks/useUnionResource/utils/fetchEntity.ts:70]: Can you confirm that this change around removing `if(!res.next)` is purely to make the code simpler or it does solve an issue?

File: src/frontend/js/hooks/useUnionResource/index.ts [hunk 2]
```diff
@@ -94,7 +94,11 @@ const useUnionResource = <
 
   // to force execution of useEffect::fetchNewPage(),
   // reset need to generate a uniq key that is part of it's dependencies.
-  const reset = () => {
+  const reset = (eofKey?: string) => {
```
Comment (NathanVss) [inline src/frontend/js/hooks/useUnionResource/index.ts:97]: What I suggest is to not have a param passed to the `reset` function, to make sure it stay simple. And to reset both list each time an invalidation occur, WDYT? After all we can consider that if there is one invalidation we should reset the entire list as they nearly the same entity

**Review by NathanVss — APPROVED** [review]

File: src/frontend/js/hooks/useUnionResource/utils/fetchEntity.ts (hunk 1 above)
Comment (rlecellier) [reply src/frontend/js/hooks/useUnionResource/utils/fetchEntity.ts:70]: I don't remember it fixing a bug. But i felt better having `eofRef.current` up to date all along on debuging and i saw no reason not to keep it like this at the end.

Comment (NathanVss) [reply src/frontend/js/hooks/useUnionResource/utils/fetchEntity.ts:70]: Allright

File: src/frontend/js/hooks/useUnionResource/index.ts (hunk 2 above)
Comment (NathanVss) [reply src/frontend/js/hooks/useUnionResource/index.ts:97]: Ok deal if it works well

**Pull request merged**